front/src/modules/timetableItem/components/ImportTimetableItem/helpers/parseXML.ts

```diff
@@ -72,6 +90,169 @@ export const getMostFrequentInterval = (schedules: TrainSchedule[]): Duration =>
+      matchedImportedSchedule: null,
+    });
+  }
+
+  const matchedImportedScheduleKeys = new Set<string>();
+  const getScheduleKey = (s: TrainSchedule) => `${s.start_time}`;
+
+  // Match OSRD Default Occurrences to imported Trains
+  osrdDefaultOccurrences.forEach((osrdOccurrence) => {
+    let bestCandidate: TrainSchedule | null = null;
+    let minTimeDifference = Infinity;
+
+    const availableImportedSchedules = sortedImportedSchedules.filter(
+      (schedule) => !matchedImportedScheduleKeys.has(getScheduleKey(schedule))
+    );
+
+    availableImportedSchedules.forEach((importedSchedule) => {
+      const importedDepartureTime = new Date(importedSchedule.start_time);
+      const timeDifferenceMinutes = Math.abs(
+        Duration.subtractDate(osrdOccurrence.startTime, importedDepartureTime).total('minute')
+      );
+
+      if (
+        timeDifferenceMinutes <= intervalMinutes / 4 &&
+        timeDifferenceMinutes < minTimeDifference
+      ) {
+        bestCandidate = importedSchedule;
+        minTimeDifference = timeDifferenceMinutes;
+      }
+    });
+
+    if (bestCandidate) {
+      osrdOccurrence.matchedImportedSchedule = bestCandidate;
+      matchedImportedScheduleKeys.add(getScheduleKey(bestCandidate));
+    }
+  });
+
+  const exceptions: PacedTrainException[] = [];
+  osrdDefaultOccurrences.forEach((osrdOccurrence, index) => {
+    if (!osrdOccurrence.matchedImportedSchedule) {
+      // Case 1: OSRD default occurrence has no correspondence in the imported file
+      // Create a disabled exception
+      exceptions.push({
+        key: uuidV4(),
+        occurrence_index: index,
+        disabled: true,
+      });
+    } else {
+      // Case 2: OSRD default occurrence has a correspondence in imported file
+      const baseException = generatePacedTrainException(
+        osrdOccurrence.matchedImportedSchedule,
+        originalPacedTrain,
+        index
+      );
+      const cleanException = omit(baseException, 'train_name');
+
+      if (Object.keys(cleanException).length > 0) {
+        const exception: PacedTrainException = {
+          ...cleanException,
+          key: uuidV4(),
+          occurrence_index: index,
+        };
+
+        exceptions.push(exception);
+      }
+    }
+  });
+
+  // Processing Unattributed Trains as added exceptions
+  const allImportedScheduleKeys = new Set(sortedImportedSchedules.map(getScheduleKey));
+  const unmatchedImportedScheduleKeys = new Set(
+    Array.from(allImportedScheduleKeys).filter((key) => !matchedImportedScheduleKeys.has(key))
+  );
+
+  sortedImportedSchedules.forEach((importedSchedule) => {
+    if (!unmatchedImportedScheduleKeys.has(getScheduleKey(importedSchedule))) {
+      return;
+    }
+
+    const updatedOccurrence = trainScheduleToPacedTrain(
+      importedSchedule,
+      pacedTrainId,
+      intervalDuration,
+      totalTimeWindow
+    );
+
+    const baseException = generatePacedTrainException(updatedOccurrence, originalPacedTrain, null);
```

Comment:
Do you really need to build a pacedTrain here ? What about directly pass the importedSchedule ?
```suggestion
    const baseException = generatePacedTrainException(importedSchedule, originalPacedTrain, null);
```